plugins/sf2_player/sf2_player.cpp

```diff
@@ -780,6 +784,13 @@ void sf2Instrument::deleteNotePluginData( NotePlayHandle * _n )
 									// do it here
 	{
 		noteOff( pluginData );
+		if( m_playingNotes.indexOf( _n ) >= 0 )
+		{
+			m_playingNotesMutex.lock();
+			m_playingNotes.remove( m_playingNotes.indexOf( _n ) );
```

Comment:
Although, maybe that's not a race condition that you're trying to protect against (as the other note removals don't have these checks). Though you still should lock `m_playingNotes` before calling `indexOf` in case another thread causes the vector to be resized before `indexOf` returns. So in either case, the locking should be moved to outside the conditional.